agent/rpc/middleware/interceptors.go

```diff
@@ -49,7 +50,11 @@ func NewRequestRecorder(logger hclog.Logger, isLeader func() bool, localDC strin
 }
 
 func (r *RequestRecorder) Record(requestName string, rpcType string, start time.Time, request interface{}, respErrored bool) {
-	elapsed := time.Since(start).Milliseconds()
+	elapsedNs := time.Since(start)
+	elapsed := float64(elapsedNs.Milliseconds())
+	if elapsed == 0 {
+		elapsed = math.Round(float64(elapsedNs.Microseconds())) / math.Pow(10, 3)
```

Comment:
IIUC, you were asking about the behavior of `.Round` not actually rounding in the PR description.

I think for Round to work like we want to it would have to look like:

```
math.Round(float64(elapsedNs.Microseconds()) / math.Pow(10, 3))
```

but if we did that then the answer is always 0 per the spec:

https://pkg.go.dev/math#Round